spanner/value.go

```diff
@@ -3631,6 +3775,11 @@ func encodeValue(v interface{}) (*proto3.Value, *sppb.Type, error) {
 			}
 		}
 		pt = protoEnumType(protoEnumfqn)
+	case NullEnum:
+		if v.Valid {
+			return encodeValue(v.EnumVal)
+		}
+		pt = protoEnumType("")
```

Comment:
We need to double check in DML, after parameter binding is available, if a parameter of type "" can be written into a column with type, "test.UserInfo", which essentially is a type mismatch.

You can also verify that internally.